project/yxuan0329/proposal.rst

```diff
@@ -0,0 +1,171 @@
+applications, particularly in spatial information systems such as maps. 
+However, beyond spatial information, data in map system should also take 
+temporal feature in considerations for better searching performance. When 
+searching nearest location on map system, users should be able to see different 
+sorting results depending on their specified searching time. In other words, 
+the users should be able to receive different searching results for the same 
+query at different days and times.
+
+The goal of this project is to integrate temporal data into the indexing 
+process, effectively creating a new temporal dimension within the R-tree 
+structure. We aspire to enable users to not only search by spatial attributes 
+but also to incorporate temporal considerations into their queries. 
+Furthermore, this project seeks to refine the organization of data within the 
+R-tree, prioritizing results based on temporal relevance, thereby delivering a 
+more comprehensive and responsive mapping experience.
+
+
+
+Prospective Users
+=================
+
+.. Describe the users of your software and how they will use it.  It is OK to
+.. combine this section with the previous one (`Problem to solve`_).
+
+This project aims to provide to users that want to search multidimensional data 
+including spatial and temporal features.
+
+System Architecture
+===================
+
+.. Analyze how your system takes input, produces results, provide interface, and
+.. performs any other operations.  Describe the system's work flow.  You may
+.. consider to use a flow chart but it is not required.  Specify the constraints
+.. assumed in your system.  Describe the modularization of the system.
+(Will be finished soon after discussion)
+
+
+Implementation Plan
+===================
+
+1. Frontend design:
+The search engine will provide a simple UI with one text block for users to 
+enter query for searching, and one text block list for showing the searching 
+results.
+
+2. Datasets:
+Generate the database with about 50 map data entry by myself for validating the 
+R-tree algorithm can works correctly. 
+
+
+API Description
+===============
+
+.. Show how your system can be programmed.  You are supposed to implement the
+.. system using both C++ and Python.  Describe how a user writes a script in the
+.. system.
+(Will be finished soon after discussion)
+
+
+
+Engineering Infrastructure
+==========================
+
+.. Describe how you plan to put together the engineering system:
+
+.. 1. Automatic build system and how to build your program
+.. 2. Version control (show how you will use it)
+.. 3. Testing framework
+.. 4. Documentation
+
+.. Some of the above information should be included in the documentation in your
+.. software.
+
+.. You may use continuous integration, but it is not required.  If you use it,
+.. describe how it works in your code development.
+
+1. Automatic build system: CMake
+2. Version control: git
+3. Testing framework: Pytest
```

Comment:
It could help your planning to write done more about what you want to test and where you will get the test data.

Reply:
No problem, I just added it.

Reply:
Please do not resolve.  It makes tracking more difficult.